### DISCUSSION TITLE
Unable to install rpm on M1 Mac

### DISCUSSION THREAD

**Discussion by anhsirk2104** [Homebrew/discussions#1905] [Getting started] [answered]
Hi,

Running into issues while installing rpm using brew. Any pointers to get this resolved would be helpful.

```
Mac-mini-2 Downloads % brew install --build-from-source rpm
==> Downloading http://ftp.rpm.org/releases/rpm-4.16.x/rpm-4.16.1.3.tar.bz2
Already downloaded: /Users/.../Library/Caches/Homebrew/downloads/490bdabcf09ad0ae5e0dc4820a86faad5112c2470621cb3153e74a4003d3776c--rpm-4.16.1.3.tar.bz2
==> ./configure --prefix=/opt/homebrew/Cellar/rpm/4.16.1.3 --localstatedir=/opt/homebrew/var --sharedstatedir=/opt/homebrew/var/lib --sysconfdir=/opt/homebrew/etc --with-path-magic=/opt/homebrew
==> make install
Last 15 lines from /Users/krishna/Library/Logs/Homebrew/rpm/02.make:
/bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H   -I.. -I.. -I../include -I../misc -I/opt/homebrew/opt/lua/include/lua -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include  -g -O2 -c -o fts.lo fts.c
/bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H   -I.. -I.. -I../include -I../misc -I/opt/homebrew/opt/lua/include/lua -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include  -g -O2 -c -o fnmatch.lo fnmatch.c
libtool: compile:  clang -DHAVE_CONFIG_H -I.. -I.. -I../include -I../misc -I/opt/homebrew/opt/lua/include/lua -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include -g -O2 -c fts.c  -fno-common -DPIC -o .libs/fts.o
libtool: compile:  clang -DHAVE_CONFIG_H -I.. -I.. -I../include -I../misc -I/opt/homebrew/opt/lua/include/lua -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include -g -O2 -c fnmatch.c  -fno-common -DPIC -o .libs/fnmatch.o
fts.c:1120:16: error: variable has incomplete type 'struct stat64'
        struct stat64 sb;
                      ^
fts.c:1120:9: note: forward declaration of 'struct stat64'
        struct stat64 sb;
               ^
1 error generated.
make[2]: *** [fts.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [install-recursive] Error 1
make: *** [install] Error 2

```

**Comment by SMillerDev** [comment, marked as answer]
Why are you building from source?

**Reply by anhsirk2104** [reply]
Doesn't allow me to install otherwise

`Downloads % brew install rpm                    
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 3 formulae.

Error: rpm: no bottle available!
You can try to install from source with:
  brew install --build-from-source rpm
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.`

**Reply by SMillerDev** [reply]
> Please note building from source is unsupported. You will encounter build failures with some formulae. If you experience any issues please create pull requests instead of asking for help on Homebrew's GitHub, Twitter or any other official channels.

This is an important part here. We can't help you because if homebrew doesn't bottle something, it means we can't get it to compile either.

**Reply by anhsirk2104** [reply]
Acknowledged.

**Comment by carlocab** [comment]
We don't support building from source, sorry.

The fact that there is no ARM bottle for this also means that we don't know how to install this on an M1 Mac.

**Comment by MagicTmac1** [comment]
How to solve the problem?

**Reply by SMillerDev** [reply]
Submit a patch to the developers of RPM that fixes this. Alternatively, create an issue with the developers of RPM (if they don't have one already).
